File: 1502-construct-k-palindrome-strings/1502-construct-k-palindrome-strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        int n = s.length(), count = 0;
        if(n == k) return true;
        if(n < k) return false;

        int mp[26] = {0};
        for(int i=0; i<n; i++) {
            mp[s[i]-'a']++;
        }
        for(int i=0; i<26; i++) {
            if(mp[i] & 1 == 1) {
                count++;
            }

        }
        if(count > k) return false;
        return true;
    }
};

File: 1502-construct-k-palindrome-strings/1502-construct-k-palindrome-strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        int countOdd = 0, n = s.length();
        if (n < k) return false;

        int[] freq = new int[26];
        char[] arr = s.toCharArray();
        for(char ch : arr) {
            freq[ch - 'a']++;
        }
        for(int i = 0; i<26; i++) {
            countOdd += (freq[i] & 1);
        }
        if(countOdd > k) return false;
        return true;
    }
}